test/test_cost_functions.jl

```diff
@@ -11,3 +11,80 @@ res = [1.0, 0.1, 0.1]
 for (ix, v) in enumerate(bps)
     @test isapprox(v, res[ix])
 end
+
+@testset "Test MarketBidCost with Polynomial Cost Timeseries with Service Forecast " begin
+    initial_time = Dates.DateTime("2020-01-01")
+    resolution = Dates.Hour(1)
+    other_time = initial_time + resolution
+    name = "test"
+    horizon = 24
+    service_data = Dict(initial_time => ones(horizon), other_time => ones(horizon))
+    polynomial_cost = repeat([(999.0, 1.0)], 24)
+    data_polynomial =
+        SortedDict(initial_time => polynomial_cost, other_time => polynomial_cost)
+    sys = create_rts_system()
+    generators = collect(get_components(ThermalStandard, sys))
+    generator = get_component(ThermalStandard, sys, get_name(generators[1]))
+    market_bid = MarketBidCost(nothing)
```

Comment:
Could you allow
```
market_bid = MarketBidCost()
```
?  It feels odd to me to make the user pass `nothing`.

Reply:
This is only for the test, I don't expect for the user to call this constructor. I saw the null constructor being used in other tests as well, so followed that pattern. This doesn't make much sense as an example as the user's would create the generator with the MarketBidCost but here I use the generator from RTS system and replace the ThreePartCost with MarketBidCost but I'm open to populate it with some values .